Update deletion notes (#13703)
PostHog/posthog.com: contents/docs/privacy/data-storage.mdx

Reviewer: Should we snippet these?

### Right to be forgotten

> <CalloutBox icon="IconWarning" title="Reusing a deleted distinct_id" type="caution">

Reviewer: ```suggestion
<CalloutBox icon="IconWarning" title="Avoid reusing deleted distinct IDs" type="caution">
```

> User deletion requests are not immediate. Instead, they are processed asynchronously. We recommend that you _avoid_ reusing a deleted `distinct_id` value for a new user, as it may lead to unexpected results while the deletion is being processed. Once the deletion process is fully processed, PostHog does not retain any data associated with the deleted `distinct_id`. 

Reviewer: ```suggestion
User deletion requests are not immediate, but processed asynchronously. We recommend that you _avoid_ reusing a deleted `distinct_id` value for a new user, as it may lead to unexpected results while the deletion is being processed. Once the deletion process is fully processed, PostHog does not retain any data associated with the deleted `distinct_id`. 
```
